expo/expo
# [docs] Remove page higher order component to fix memory issues

# Why

My best guess is that this document layout is being fully transpiled on all pages. Even though we should have caching enabled, it does seems to retranspile this (and all nested dependencies within that page). Unforutnately, because the debugging tooling around this is also failing due to memory issues, I can't really confirm that this is happening.

This change moves us away from the higher order component and the custom webpack loader. The front-matter tasks within the custom loader are now handled by remark's plugins and AST modifications.

> The current table of contents system needs to be fixed, it currently does not work.

# How

_See commits_

# Test Plan

- Check if you can build the docs without `NODE_OPTIONS=--max-old-space-size=5120`
- Check if CI can build the docs without that property
- Check if result is still the same after exporting

# Checklist
